### PR TITLE
dot/core, lib/grandpa: connect grandpa to core, networking, begin service on startup

### LINKED ISSUE #933: grandpa: convert VoteMessage, Justification, and FinalizationMessage to network messages
## Expected Behavior

- add VoteMessage, Justification, and FinalizationMessage to network messages
- have them implement the `network.Message` interface
- update grandpa channels to have type `network.Message`

## Current Behavior

- grandpa messages are in grandpa package
- `in`, `out` chans only use `VoteMessage`

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #934: grandpa: broadcast FinalizationMessage upon block finalization
## Expected Behavior

- see `grandpa.playGrandpaRound` and `grandpa.attemptToFinalize`
- requires #933 
- also need to implement justification collecting

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #943: grandpa: connect to networking layer
## Expected Behavior

- need to send and receive grandpa VoteMessages, FinalizationMessages, and JustificationMessages through the networking layer

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by noot**
## Changes

- add `createGRANDPAService` to dot, start grandpa service on startup
- add finality message processing to core

## Tests

```
go test ./... -short
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- closes #943

### CODE REVIEW

File: dot/core/finality.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import (
+	"github.com/ChainSafe/gossamer/dot/network"
+
+	log "github.com/ChainSafe/log15"
+)
+
+// processConsensusMessage routes a consensus message from the network to the finality gadget
+func (s *Service) processConsensusMessage(msg *network.ConsensusMessage) error {
+	in := s.finalityGadget.GetVoteInChannel()
+	fm, err := s.finalityGadget.DecodeMessage(msg)
```
Comment (ryanchristo) [inline dot/core/finality.go:28]: I might have overlooked this previously but should `DecodeMessage` be an instance method `Decode` for the `ConsensusMessage` type and included with the other network messages or is the idea to keep grandpa modular (and then maybe we should take into consideration when investigating #885)?

File: dot/core/finality.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+// sendFinalityMessages routes a FinalizationMessage from the finality gadget to the network
+func (s *Service) sendFinalizationMessages() {
+	out := s.finalityGadget.GetFinalizedChannel()
+	for v := range out {
+		// TODO: safety
+		// TODO: update state.finalizedHead
+		log.Debug("[core] sending FinalityMessage to network", "msg", v)
+		log.Info("[core] finalized block!!!", "msg", v)
```
Comment (ryanchristo) [inline dot/core/finality.go:62]: !!!

File: lib/keystore/helpers.go [hunk 3]
```diff
@@ -127,30 +127,44 @@ func LoadKeystore(key string) (*Keystore, error) {
 
 	if key != "" {
 
-		kr, err := NewSr25519Keyring()
+		srkr, err := NewSr25519Keyring()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create keyring: %s", err)
+		}
+
+		edkr, err := NewEd25519Keyring()
```
Comment (ryanchristo) [inline lib/keystore/helpers.go:135]: :100:

File: dot/core/finality.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		s.msgSend <- msg
+	}
+}
+
+// sendFinalityMessages routes a FinalizationMessage from the finality gadget to the network
+func (s *Service) sendFinalizationMessages() {
+	out := s.finalityGadget.GetFinalizedChannel()
+	for v := range out {
+		// TODO: safety
+		// TODO: update state.finalizedHead
```
Comment (ryanchristo) [inline dot/core/finality.go:60]: ?

Comment (noot) [reply dot/core/finality.go:60]: this is updated here: https://github.com/ChainSafe/gossamer/pull/953

**Pull request merged**